Add "People" page to list contributors who are available for services

anaulin opened the pull request:
Very rough first cut at the "People" page, as outlined in https://github.com/zinc-collective/www.zinc.coop/issues/71

To do:
- [x] improve spacing between people
- [x] figure out why images are not showing up as square
- [x] align name vertically with top of image
- [x] make link list markers look like in the mocks (with `>`)
- [x] add membership type field
- [x] add specialties field
- [x] limit maximum content width so that it doesn't exceed ~670px
- [x] add actual members' content

Screenshot of how it looks right now:
![localhost_8080_people_](https://user-images.githubusercontent.com/6729309/109596521-4638a800-7acb-11eb-907f-0200b66e89ee.png)

anaulin committed:
Run npm audit fix

anaulin committed:
Start "People" page to list people's bios and skills.

anaulin commented:
@colombene @zspencer fyi showing my work as I go. Needs more styling, but starting to take shape.

@zspencer or anyone else reading this, feel free to add commits to this draft PR if you the spirit moves you to help out. 😄

zspencer commented:
Very nice progress! I appreciate it!

Is there an intent to populate the content from the Airtable spreadsheet? That's something Eleventy is supposed to do "easily." (lol, anything in computers happening "easily")

anaulin commented:
For this first pass, I was thinking we would just populate the data manually. I don't know who in that Airtable is still interested in being listed here, if their info is up-to-date, etc.

If we like the page and more people want to be on it, then we can investigate some sort of automation to import the data from Airtable.

anaulin committed:
Add container for each person and vertical margin between them.

anaulin committed:
Add explicit width and height to person image.

anaulin committed:
Align person name vertically with top of image.

anaulin committed:
Add > list marker.

anaulin committed:
Add membership type, interests and specialties fields.

anaulin committed:
Add a little bit more of vertical top space to each person.

anaulin commented:
Alrighty, I think this is ready to add in our initial set of real people content! See the screenshot in PR description for an idea of how this looks right now. Feedback on my HTML/CSS implementation is also welcome (I have a `<br/>` suggestions for improving that without having to add yet another `div` welcome).

@user512 @zspencer @colombene If you want to be listed on this page, please give me your blurbs, links and headshot images and I'll add them to this PR. The fields each person needs to fill in are:
- square headshot image
- membership type (member or maintainer)
- name
- location
- one or more links; each link has a label (e.g. "website" or "linkedin") and a URL
- specialties
- interests
- bio

anaulin committed:
Add content for Ana Ulin.

anaulin commented:
I've added example content for myself, you can see it in the screenshot above.

anaulin committed:
Move people content into separate JSON data file.

anaulin committed:
Improve spacing between image and text.

zspencer commented on src/people/index.md:
```diff
@@ -0,0 +1,25 @@
+  <p><strong>Specialties:</strong> <em>{{ person.specialties }}</em></p>
+  <p><strong>Interests:</strong> <em>{{ person.interests }}</em></p>
+  <p>{{ person.bio }}</p>
+</div>
```
The current object hierarchy is:

```
- .person-container
  - .person-card
    - img
    - div
      - h3
      - ul
        -li 
  - p
  - p 
```

Maybe something like this would reveal the intent a bit better?

```
.person
  h3.name
  img.photo alt={{ person.name }}
  main
    p.bio
    ul.links
      li
  footer
    h4 Specialties # Note: Can style both the h4 and the p to be an inline-block or inline-flex so they stay on the same line together while preserving the information hierarchy
    p
    h4 Interests
    p
```
It'd be even fancier if we conformed to the [hCard format for making the data parsable](http://microformats.org/wiki/h-card)

anaulin replied:
I've made the markup a little bit closer to the intent while preserving the same look. It tries to strike some balance between a cleaner markup and minimize the amount of fancy bending-over-backward CSS trickery to make things look "just so" while keeping the markup strictly semantic.

There might be some simple or obvious CSS tricks that I'm missing that would make this even better, feel free to suggest if you see anything like that. That said, I am skeptical of the value of doing too much to make the markup perfectly semantic at the expense of increased CSS complexity.

zspencer replied:
Dig it!

zspencer reviewed (approved):
I like where this is going, I have some preference for making the HTML structure a more direct reflection of the information it's presenting, but I don't think it's blocking.

anaulin committed:
Make markup a little bit more intent-revealing. Add alt to image.

anaulin commented:
@zinc-collective/contributors if you are interested in being listed on Zinc's new public "People" page (see screenshot above), please respond here and provide the info needed to fill in your card:
* name
* square headshot image
* membership type (member or maintainer)
* location
* one or more links; each link has a label (e.g. "website" or "linkedin") and a URL
* specialties
* interests
* bio

I'll give folks a week or so to respond, and then I'll merge this with whatever information we have. We can always add/edit/remove folks later, of course.

user512 commented:
This looks great, maybe we can post this https://github.com/zinc-collective/www.zinc.coop/pull/72#issuecomment-786389003 on Discord so non-developer can also provide services?

colombene commented:
This is great! So exciting. :) Working on my info now. Only design note - could you shorten the line length so it doesn't go full width? For reference, Medium.com maxes out about 670px.
FYI, I'm planning to simplify the design (header/footer/color). What I did previously seems more appropriate for a venture than the collective as a whole. Not sure if that's relevant because it doesn't look applied, but wanted to note.

zspencer commented:
Name: Zee Spencer
Square headshot image: N/A, and if our design requires people to provide an image we may want to adjust the design to play nicely with folks who are not comfortable with showing their face. (Maybe @colombene has insights here). 
membership type: Member
Location: San Francisco, CA
Links:
  - Home: https://www.zeespencer.com
  - Professional Social (Twitter): https://twitter.com/zspencer
  - Professional Social (LinkedIn): https://www.linkedin.com/in/zspencer/
  - Personal Social: https://wandering.shop/@zee
Specialties:
  - Cooperative Finance and Operations
  - Software Architecture, Testing, and Engineering
Interests:
  - Reading and Story-Telling
  - Adventure and Story-Telling Games
Bio: Zee has worked in software for nearly two decades, primarily in engineering organizations. Zee has woven his career between individual contributor, management, leadership and training roles. At present, his focus is on developing collectively organized workplaces, with an emphasis on employee ownership of software companies.

anaulin commented:
> This looks great, maybe we can post this [#72 (comment)](https://github.com/zinc-collective/www.zinc.coop/pull/72#issuecomment-786389003) on Discord so non-developer can also provide services?

Good idea. Done. (I also emailed the maintainers and contributors mailing lists.)

anaulin commented:
Good point on making the headshot optional, @zspencer. I'll work that in whenever I work on adding your info into the PR.

I'll go with allowing the name/location/links block to float to the left margin if there is no image. @colombene, let me know if you have an opinion on how to deal with this case.

anaulin commented:
> Only design note - could you shorten the line length so it doesn't go full width? For reference, Medium.com maxes out about 670px.

I'll look into this when I work on this PR next. I'll probably look into applying that generally to the entire site's "main content" area. (Let me know if that's not right!)

> FYI, I'm planning to simplify the design (header/footer/color). What I did previously seems more appropriate for a venture than the collective as a whole. Not sure if that's relevant because it doesn't look applied, but wanted to note.

I wanted to separate the work of adding a new page from doing broader restyling that would affect the site as a whole, so I didn't change colors, header, or footer. It is easier to talk about and handle in separate PRs, I think, as well as dividing the work into more manageable steps.

That said, if there is styling specific to this page that I missed, please let me know, @colombene.

zspencer commented:
@colombene's not re: Width reminds me of:

- Design-wise: https://graphicdesign.stackexchange.com/questions/13724/recommended-column-width-for-text-reading-digital-vs-printed
- Engineering-wise: https://web.dev/min-max-clamp/

cjerozal commented:
Thanks for putting this together @anaulin!

So the primary purpose of this page is to show who is available for service work? If so, I think we should add something saying so. However, I think it could be cool to let anyone seriously involved with Zinc be on the page if they want, but have an individual indicator for whether or not one is available for service work.

As a former member and current maintainer I'm definitely biased on this, but... As a primarily externally facing page, I don't think we should include member vs maintainer info. It's confusing enough for those who are actually involved with Zinc IMO. No need for potential clients to learn about the distinction when it shouldn't impact them.

anaulin commented:
@cjerozal showcasing folks who are available for work is one of the purposes of this page, I don't know if I would say that it is the primary purpose. Whoever wants to be listed on this page is welcome to it, not just people who are available for work. There is a planned update to the "Work/Products" area, which will crosslink to this page and will have some wording about some people being available for contract work. The website brief has more details on this: https://docs.google.com/document/d/17DjiqnA5ItNQPi_Qj8NfLPgFjZba2lDZyvGq-HGjGCw/edit#

anaulin committed:
Clamp width of paragraphs in main content, to max 800px.

anaulin committed:
Add Zee's info. Update Ana's info.

anaulin commented:
@colombene  I set a max width (thanks for the clamp reference, @zspencer) to the text. It will apply across the site. You can see in the update screenshot above what it looks like. Let me know what you think!
 
@zspencer I added your info, and updated the template to skip the image if there isn't one. See results in the screenshot above.

zspencer commented:
Love it! Thank you!

anaulin commented:
Looks like, for now, nobody else is interested to be listed on this page. @colombene as soon as I have your deets we can merge this in and make it live. 🚢

anaulin commented:
Seems like there isn't an immediate enthusiasm for anyone else to add themselves to this page, so if there are no objections, I'll will merge this in tomorrow. We can add more people to the page as their info becomes available.

zspencer commented:
Thank you @anaulin !!!!
